core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -267,6 +267,8 @@ public TableDataMap getDataMap(CarbonTable table, DataMapSchema dataMapSchema) {
           }
         }
       }
+    } else {
+      dataMap.clear();
```

Comment:
why is it required? please add comment

Reply:
I think it loses the purpose of cache if you clear for every retrieval. clear should be called only flush the cache. why do you need to flush cache for every call?

Reply:
I move it into carbonReader close method